xopt/generators/bayesian/bax/acquisition.py

```diff
@@ -0,0 +1,135 @@
+import torch
+from botorch.acquisition.analytic import AnalyticAcquisitionFunction
+from botorch.models.model import Model
+from botorch.utils.transforms import t_batch_mode_transform
+from torch import Tensor
+
+from xopt.generators.bayesian.bax.algorithms import Algorithm
+
+
+class ExpectedInformationGain(AnalyticAcquisitionFunction):
+    r"""Single outcome expected information gain`
+        Currently only works with model = SingleTaskGP (not ModelListGP)
+    Example:
+        >>> model = SingleTaskGP(train_X, train_Y)
+        >>> EIG = ExpectedInformationGain(model, algo, algo_params)
+        >>> eig = EIG(test_X)
+    """
+
+    def __init__(self, model: Model, algo: type[Algorithm]) -> None:
+        r"""Single-outcome Expected Improvement (analytic).
+
+        Args:
+            model: A fitted single-outcome model.
+            maximize: If True, consider the problem a maximization problem.
+        """
+
+        super().__init__(model=model)
+        self.algo = algo
+        self.xs_exe, self.ys_exe = self.algo.get_exe_paths(
+            self.model
+        )  # get sampled execution paths and set self.xs_exe, self.ys_exe
+
+        self.model(
```

Comment:
instead of evaluating the model as a way to check algorithm execution paths, create a function for validating execution paths. Function should check tensor shape, etc. etc. required for successful use in the model